### DISCUSSION TITLE
Question: Can Reaction SMARTS preserve stereochemistry?

### DISCUSSION THREAD

**Discussion by lilyminium** [rdkit/rdkit#4059] [Q&A]
Hi everyone,

I am trying to use `ReactionFromSmarts` to add a `substituent` structure onto a `reactant` structure with stereochemistry encoded. However, stereochemistry is not present in the product upon running the reaction with `RunReactants`. Is there a way to preserve this stereochemistry?

My code is below:

```python
>>> from rdkit import Chem
>>> from rdkit.Chem import rdChemReactions
>>> reactant = (Chem.MolFromSmiles("C(=C(/C([2*]))\O)(/[1*])"),)  # stereochemistry encoded in [1*]

>>> substituent = "[*:1][CH2][C](=[CH2])[OH]"  # no stereochemistry

>>> rxn = rdChemReactions.ReactionFromSmarts(f"[*:1]-[1*]>>{substituent}")
>>> products = rxn.RunReactants(reactant)
>>> Chem.MolToSmiles(products[0][0], isomericSmiles=True, allHsExplicit=True)
'[2*][CH2][C]([OH])=[CH][CH2][C](=[CH2])[OH]'  # no stereochemistry around =[CH]
```

I am running on
* MacOS Catalina (10.15.7)
* RDKit version 2021.03.1
* Python 3.8.8

**Comment by greglandrum** [comment]
@lilyminium: this one is tricky. I'm going to have to play around for a bit to see if I can come up with a way to handle it
